internal/job/job.go

```diff
@@ -0,0 +1,223 @@
+package job
+
+import (
+	"context"
+	"errors"
+	"io"
+	"os"
+	"os/exec"
+	"sync"
+	"sync/atomic"
+
+	"github.com/joshuarubin/teleport-job-worker/internal/config"
+	"github.com/joshuarubin/teleport-job-worker/internal/safebuffer"
+	"github.com/joshuarubin/teleport-job-worker/internal/user"
+)
+
+// Job represents a system process
+type Job struct {
+	cfg    *config.Config
+	id     ID
+	userID user.ID
+	cmd    *exec.Cmd
+	buf    *safebuffer.Buffer
+
+	startOnce sync.Once
+
+	stopOnce sync.Once
+	stopErr  atomic.Value
```

Comment:
does this need to be atomic?

Reply:
If `Stop()` is called concurrently, one will "win" and execute the `Once` func and the other will read the value of `stopErr` while the `Once` func may be writing it. So yes, it does need coordination, but there is still a bug. In the case I described, the "loser" that doesn't run the `Once` func may erroneously return a nil error because it read the value before the `Once` func hasn't yet written it. It probably needs a `RWMutex` instead.

Reply:
Scratch what I just said. I forgot how Once works. Concurrent calls to `Once.Do()` all block until the "one" func running returns, so there isn't a concurrency issue here. Will make it a var.